src/pydvl/value/shapley/montecarlo.py

```diff
@@ -152,6 +148,146 @@ def permutation_montecarlo_shapley(
     return map_reduce_job()
 
 
+def permutation_montecarlo_classwise_shapley(
```

Comment:
I would move this function to `classwise.py` and make it private. Why would a user need it? And if it is indeed needed then the name has to change because this is the same as that of functions that compute actual shapley values, whereas this is conditioned on a class. At any rate, this belongs in `classwise.py`